core/runtime.rs

```diff
@@ -622,6 +594,8 @@ impl JsRuntime {
     js_runtime.init_extension_ops().unwrap();
     let realm = js_runtime.global_realm();
     js_runtime.init_extension_js(&realm).unwrap();
+    *loaded_extensions.borrow_mut() = true;
+    js_runtime.module_map.as_ref().unwrap().borrow_mut().loader = loader;
```

Comment:
After extensions are initialised, this unsets `ExtModuleLoader` in favour of the user's loader. I think we should have a concrete `CoreModuleLoader` which always wraps the user's loader. Then we can declare it concretely instead of stacking two `dyn ModuleLoader`. That way we can remove `resolve_helper()` and not have to pass around `loaded_extensions`.